node/node.go

```diff
@@ -0,0 +1,289 @@
+	// provided, the 'omit' flag for that module must be set to false (which is
+	// the default setting).
+	ConsensusSet    modules.ConsensusSet
+	Explorer        modules.Explorer
+	Gateway         modules.Gateway
+	Host            modules.Host
+	Miner           modules.TestMiner
+	Renter          modules.Renter
+	TransactionPool modules.TransactionPool
+	Wallet          modules.Wallet
+
+	// The high level directory where all the persistence gets stored for the
+	// moudles.
+	Dir string
+}
+
+// Node is a collection of Sia modules operating together as a Sia node.
+type Node struct {
+	// The modules of the node. Modules that are not initialized will be nil.
+	ConsensusSet    modules.ConsensusSet
+	Explorer        modules.Explorer
+	Gateway         modules.Gateway
+	Host            modules.Host
+	Miner           modules.TestMiner
+	Renter          modules.Renter
+	TransactionPool modules.TransactionPool
+	Wallet          modules.Wallet
+
+	// The high level directory where all the persistence gets stored for the
+	// moudles.
+	Dir string
+}
+
+// Close will call close on every module within the node, combining and
+// returning the errors.
+func (n *Node) Close() (err error) {
+	if n.Explorer != nil {
+		err = errors.Compose(n.Explorer.Close())
+	}
+	if n.Miner != nil {
+		err = errors.Compose(n.Miner.Close())
+	}
+	if n.Host != nil {
+		err = errors.Compose(n.Host.Close())
+	}
+	if n.Renter != nil {
+		err = errors.Compose(n.Renter.Close())
+	}
+	if n.Wallet != nil {
+		err = errors.Compose(n.Wallet.Close())
+	}
+	if n.TransactionPool != nil {
+		err = errors.Compose(n.TransactionPool.Close())
+	}
+	if n.ConsensusSet != nil {
+		err = errors.Compose(n.ConsensusSet.Close())
+	}
+	if n.Gateway != nil {
+		err = errors.Compose(n.Gateway.Close())
+	}
+	return err
+}
+
+// New will create a new test node. The inputs to the function are the
+// respective 'New' calls for each module. We need to use this awkward method of
+// initialization because the siatest package cannot import any of the modules
+// directly (so that the modules may use the siatest package to test
+// themselves).
+func New(params NodeParams) (*Node, error) {
+	dir := params.Dir
+
+	// Gateway.
+	g, err := func() (modules.Gateway, error) {
+		if params.CreateGateway && params.Gateway != nil {
+			return nil, errors.New("cannot both create a gateway and use a passed in gateway")
+		}
+		/* Template for dealing with optional dependencies:
+		if !params.CreateGateway && parames.GatewayDependencies != nil {
+			return nil, errors.New("cannot pass in gateway dependencies if you are not creating a gateway")
+		}
+		*/
+		if params.Gateway != nil {
+			return params.Gateway, nil
+		}
+		if !params.CreateGateway {
+			return nil, nil
+		}
+		/* Template for dealing with optional dependencies:
+		if params.GatewayDependencies == nil {
+			gateway.New(...
+		} else {
+			gateway.NewDeps(...
+		}
+		*/
+		return gateway.New("localhost:0", false, filepath.Join(dir, modules.GatewayDir))
```

Comment:
we will need a way to specify this, as well as the host address and the `bootstrap` bool. Maybe just add them to `NodeParams`.

Reply:
Ah yeah. I think a good way to do this would just have a bunch of fields for each module. We can add them as we need them.